doc/source/ray-core/examples/rdt/grpo_contextual_bandits.py

```diff
@@ -0,0 +1,432 @@
+"""
+Reinforcement learning example using GPU-to-GPU Ray Direct Transport (RDT) and GRPO algorithm.
+"""
+import argparse
+import copy
+import threading
+from contextlib import contextmanager
+from typing import Any
+
+import ray
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.distributions import Categorical, kl_divergence
+from tqdm.auto import tqdm
+
+
+STATE_DIM = 2  # Contextual bandit in 2D
+# AKA 'group size'. Eight compass directions: [W, NW, N, NE, E, SE, S, SW]
+ACTION_DIM = 8
+BASE_LR = 5e-6
+ADAM_EPS = 1e-8
+EMA_DECAY = 0.999
+PPO_CLIP_EPS = 0.5
+KL_COEFF = 0.1
+BATCH_SIZE = 16
+GRAD_CLIP_NORM = 1.0
+
+# Unit direction vectors for 8 compass actions (W, NW, N, NE, E, SE, S, SW)
+diag = 2**0.5 / 2.0
+action_dirs = torch.tensor(
```

Comment:
If this is a constant, shall we make the name all caps?

Let's also spell out the name. Is dirs short for directions?

Reply:
Good idea, done